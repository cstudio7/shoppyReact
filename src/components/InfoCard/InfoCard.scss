$color-primary-light: #7ed56f;
$color-primary-dark: #28b485;
$color-secondary-light: #ffb900;
$color-secondary-dark: #ff7730;
$color-tertiary-light: #2998ff;
$color-tertiary-dark: #5643fa;

.card {
  width: 90%;
  margin-left: 5%;
  margin-bottom: 25%;
  position: relative;
  height: 10em;
  perspective: 150em;

  &__side {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 25em;
    backface-visibility: hidden;
    transition: all 0.6s ease;
    box-shadow: 1em 1em 2em rgba(0.2, 4, 0, 0.2);

    &--front {
      background-image: linear-gradient(
        to right bottom,
        $color-primary-light,
        $color-primary-dark
      );

      &-1 {
        background-image: linear-gradient(
          to right bottom,
          $color-primary-light,
          $color-primary-dark
        );
        height: 50%;
      }

      &-2 {
        background-image: linear-gradient(
          to right bottom,
          $color-tertiary-light,
          $color-tertiary-dark
        );
        height: 50%;
      }

      &-3 {
        background-image: linear-gradient(
          to right bottom,
          $color-secondary-light,
          $color-secondary-dark
        );
        height: 50%;
      }
    }
  }

  &__description {
    text-transform: uppercase;
    color: #fff;
    font-size: 2rem;
    padding: 1rem 0;
    width: 80%;
    margin-left: 20%;
  }

  .infoCard.ui.card {
    z-index: -2;
    margin-left: 1%;
    width: 98%;
  }

  .numberOfItems {
    font-size: 3rem;
    margin: auto;
    margin-top: 3%;
  }
}

@media only screen and (min-width: 768px) {
  .card {
    margin-bottom: 15%;
  }
}

@media (min-width: 769px) {
  .card {
    width: 40%;
    margin-bottom: 10%;
  }
  .card__description {
    margin: auto;
  }
}
