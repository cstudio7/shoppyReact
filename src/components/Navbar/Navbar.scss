.mainContainer {
  height: 100%;
}
.pageContent {
  margin-top: 64px;
}

@media (min-width: 769px) {
  .pageContent {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
  }
}
