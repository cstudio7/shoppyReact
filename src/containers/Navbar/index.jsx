import React, { Component } from 'react';
import PropTypes from 'prop-types';
import NavbarComponent from '../../components/Navbar';

export default class Navbar extends Component {
  render() {
    return <NavbarComponent userRole="admin" displayPage={this.props.displayPage} />;
  }
}
Navbar.propTypes = {
  displayPage: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]).isRequired
};
