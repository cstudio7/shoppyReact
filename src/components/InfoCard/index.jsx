import React, { Component, Fragment } from 'react';
import { Card, Button, Image, Icon } from 'semantic-ui-react';
import Modal from 'react-responsive-modal';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import selectModalForm from '../../helpers/SelectModalForm';
import { deleteProductCategory } from '../../actions/deleteCategoryAction';
import { deleteProduct } from '../../actions/deleteProductAction';
import { deleteStoreAttendant } from '../../actions/deleteAttendantAction';
import { getUserInfo } from '../../helpers/jwtHelper';
import './InfoCard.scss';

class InfoCards extends Component {
  state = {
    open: false,
    deleteOpen: false
  };

  onOpenModal = () => {
    this.setState({ open: true });
  };

  onCloseModal = () => {
    this.setState({ open: false });
  };

  openDeleteModal = () => {
    this.setState({ deleteOpen: true });
  };

  closeDeleteModal = () => {
    this.setState({ deleteOpen: false });
  };

  displayAdminButtons = (userRole) => {
    if (userRole === 'admin') {
      return (
        <div className="ui two buttons adminActions">
          <Button basic color="green" onClick={this.onOpenModal}>
            Edit
          </Button>
          <Button basic color="red" onClick={this.openDeleteModal}>
            Delete
          </Button>
        </div>
      );
    }
  };

  displayAddToCartButton = (currentPage, userRole) => {
    if (currentPage === 'product' && userRole === 'attendant') {
      return (
        <Button animated>
          <Button.Content visible>Add to Cart</Button.Content>
          <Button.Content hidden>
            <Icon name="arrow right" />
          </Button.Content>
        </Button>
      );
    }
  };

  createCard = () => {
    const {
      imageUrl,
      nameToDisplay,
      description,
      mobileNumber,
      userRole,
      currentPage
    } = this.props;
    return (
      <Card className="categoryCard">
        <Image src={imageUrl} />
        <Card.Content extra>
          <Card.Header className="adminCardTitle">{nameToDisplay}</Card.Header>
          <Card.Description className="adminCardDescription">{description}</Card.Description>
          <Card.Description className="adminCardDescription">{mobileNumber}</Card.Description>
          {this.displayAdminButtons(userRole)}
          {this.displayAddToCartButton(currentPage, userRole)}
        </Card.Content>
      </Card>
    );
  };

  deleteContent = (currentPage, componentId) => {
    switch (currentPage) {
      case 'category':
        this.props.deleteProductCategory(componentId);
        break;
      case 'product':
        this.props.deleteProduct(componentId);
        break;
      case 'attendant':
        this.props.deleteStoreAttendant(componentId);
        break;
      default:
        break;
    }
  };

  displayModals = () => {
    const { open, deleteOpen } = this.state;
    const { currentPage, id } = this.props;
    return (
      <Fragment>
        <Modal open={open} onClose={this.onCloseModal} center>
          {selectModalForm(currentPage, this.props)}
        </Modal>
        <Modal open={deleteOpen} onClose={this.closeDeleteModal} center>
          <p className="confirmDelete">{`Are You Sure You want to delete this ${currentPage}`}</p>
          <Button negative className="noDelete" onClick={this.closeDeleteModal}>
            No
          </Button>
          <Button
            positive
            icon="checkmark"
            labelPosition="right"
            content="Yes"
            onClick={this.deleteContent.bind(this, currentPage, id)}
          />
        </Modal>
      </Fragment>
    );
  };

  render() {
    const userInfo = getUserInfo();
    return !userInfo ? (
      <Redirect to="/login" />
    ) : (
      <Fragment>
        {this.displayModals()}
        <div className="categoryCardContainer">{this.createCard()}</div>
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  state
});

export { InfoCards as InfoCardsPage };
export default connect(
  mapStateToProps,
  { deleteProductCategory, deleteProduct, deleteStoreAttendant }
)(InfoCards);

InfoCards.propTypes = {
  currentPage: PropTypes.string,
  imageUrl: PropTypes.string,
  nameToDisplay: PropTypes.string,
  description: PropTypes.string,
  mobileNumber: PropTypes.string,
  userRole: PropTypes.string
};

InfoCards.defaultProps = {
  currentPage: '',
  imageUrl: '',
  nameToDisplay: '',
  description: '',
  mobileNumber: '',
  userRole: ''
};
