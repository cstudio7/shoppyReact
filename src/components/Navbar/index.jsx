import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Toolbar from '../Toolbar/Toolbar';
import SideDrawer from '../SideDrawer/SideDrawer';
import Backdrop from '../Backdrop/Backdrop';
import './Navbar.scss';

class Navbar extends Component {
  state = {
    sideDrawerOpen: false
  };

  drawerToggleClickHandler = () => {
    this.setState(prevState => ({ sideDrawerOpen: !prevState.sideDrawerOpen }));
  };

  backdropClickHandler = () => {
    this.setState({ sideDrawerOpen: false });
  };

  render() {
    let backdrop;

    if (this.state.sideDrawerOpen) {
      backdrop = <Backdrop click={this.backdropClickHandler} />;
    }
    return (
      <div className="mainContainer">
        <Toolbar drawerClickHandler={this.drawerToggleClickHandler} />
        <SideDrawer show={this.state.sideDrawerOpen} />
        {backdrop}
        <main className="pageContent">{this.props.displayPage}</main>
      </div>
    );
  }
}

Navbar.propTypes = {
  displayPage: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]).isRequired
};

export default Navbar;
