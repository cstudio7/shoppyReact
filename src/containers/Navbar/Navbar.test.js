import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './index';
import InfoCard from '../../components/InfoCard';

describe('Test the Navbar Component', () => {
  beforeAll(() => {
    localStorage.setItem(
      'Authentication',
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlMb2FkIjp7ImlkIjoiN2NhM2JjYzktZTMzOS00OTcwLTk0YjMtYjI5YTg1N2ViM2ExIiwicm9sZSI6W119LCJpYXQiOjE1NTE0NjE4OTcsImV4cCI6MTU1MTU0ODI5N30.IzGf0L_pH9W_gN2AC2Ee5aWqwSO4l3fa46ZrcyUbMKU'
    );
  });
  afterEach(() => {
    localStorage.removeItem('Authentication');
  });
  it('should match snapshot', () => {
    const userRole = 'admin';
    const wrapper = shallow(<Navbar displayPage={InfoCard} userRole={userRole} />);
    expect(wrapper).toMatchSnapshot();
  });
});
