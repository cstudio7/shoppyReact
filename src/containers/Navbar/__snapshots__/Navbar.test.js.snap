// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Navbar Component should match snapshot 1`] = `
<Navbar
  displayPage={[Function]}
  userRole=""
/>
`;
